aptos-move/framework/aptos-framework/sources/object_code_deployment.move

```diff
@@ -0,0 +1,73 @@
+module aptos_framework::object_code_deployment {
+    use std::bcs;
+    use std::features;
+    use std::signer;
+    use std::string;
+    use std::vector;
+    use aptos_framework::account;
+    use aptos_framework::code;
+    use aptos_framework::code::PackageRegistry;
+    use aptos_framework::object;
+    use aptos_framework::object::{ExtendRef, Object};
+
+    /// Object code deployment not supported.
+    const EOBJECT_CODE_DEPLOYMENT_NOT_SUPPORTED: u64 = 0x1;
+    /// Not the owner of the `PublisherRef`
+    const ENOT_OWNER: u64 = 0x2;
+
+    #[resource_group_member(group = aptos_framework::object::ObjectGroup)]
+    struct PublisherRef has key {
+        extend_ref: ExtendRef,
+    }
+
+    /// Create a new object to host the code, and `PublisherRef` if the code is upgradeable,
+    /// Send `PublisherRef` to object signer.
+    public entry fun publish(
+        publisher: &signer,
+        metadata_serialized: vector<u8>,
+        code: vector<vector<u8>>,
+    ) {
+        assert!(features::is_object_code_deployment_enabled(), EOBJECT_CODE_DEPLOYMENT_NOT_SUPPORTED);
+
+        let object_seed = object_seed(signer::address_of(publisher));
+        let constructor_ref = &object::create_named_object(publisher, object_seed);
```

Comment:
Just to confirm, it is possible to publish multiple packages from the same account (but notably not in the same txn) because we include the seqnum in the seed?

Reply:
You can publish one package per transaction, each deploying to a unique object address.